persistence-test/src/main/java/io/stargate/it/PersistenceTest.java

```diff
@@ -602,6 +606,49 @@ public void testSecondaryIndexes() throws ExecutionException, InterruptedExcepti
                 .keyspace(keyspace)
                 .table(table)
                 .toString());
+
+    String indexClass = "org.apache.cassandra.index.sasi.SASIIndex";
+    Map<String, String> indexOptions = new HashMap<>();
+    indexOptions.put("mode", "CONTAINS");
+    dataStore
+        .queryBuilder()
+        .create()
+        .custom(indexClass)
+        .index("byD")
+        .ifNotExists()
+        .on(keyspace, table)
+        .column("d")
+        .options(indexOptions)
+        .build()
+        .execute()
+        .join();
+
+    Table actualTable = dataStore.schema().keyspace(keyspace).table(this.table);
+    Table expectedTable =
+        Schema.build()
+            .keyspace(keyspace)
+            .table(table)
+            .column("a", Int, PartitionKey)
+            .column("b", Varchar)
+            .column("c", Uuid)
+            .column("d", Varchar)
+            .secondaryIndex("byB")
+            .column("b")
+            .secondaryIndex("byC")
+            .column("c")
+            .secondaryIndex("byD")
+            .column("d")
+            .indexClass(indexClass)
+            .indexOptions(indexOptions)
+            .build()
+            .keyspace(keyspace)
+            .table(table);
+
+    // TODO: as indexes are stored in a Map there's no guarantee that the order they
+    // printed when calling table#toString() is the same in both expected and actual table
+    assertThat(actualTable.indexes().size()).isEqualTo(expectedTable.indexes().size());
+    assertThat(actualTable.index("byD").toString())
+        .isEqualTo(expectedTable.index("byD").toString());
```

Comment:
To run this test you need to alias 127.0.0.11 to 127.0.0.1. On MacOS this can be done with:
```
sudo ifconfig lo0 alias 127.0.0.11 up
```
Then you need to run each one of the subclasses individually: `CassandraPersistenceIT` in the 3.11 and 4.0 modules, and `DsePersistenceIT`, but you probably can't run that one.

I was able to run 3.11 and DSE successfully in my environment. For 4.0, SASI indices need to be enabled explicitly in `cassandra.yaml`, I'm looking into how to do that.

Reply:
Awesome! :)